src/core/IronPython.Modules/fcntl.cs

```diff
@@ -27,6 +35,91 @@ a file object.
         """;
 
 
+    #region  fcntl
+
+    [LightThrowing]
```

Comment:
Is the light throwing path expected to occur in normal use? If not then I would consider making a regular exception.

Similar question for the other light throwing methods below.

Reply:
It depends on the command but yes, sometimes. Those errors are very inexpensive to return so the usage pattern for some commands is simply try/error. For example handling errors `EAGAIN`, `EWOULDBLOCK`., `EACCES`. Having `LightThrowing` makes it relatively inexpensive here too. _Nota bene_ the code still throws regular exceptions for programming errors like invalid value. That's intentional so that the errors do not go unnoticed when these calls are done from C#.